src/serialization/serialization.rs

```diff
@@ -0,0 +1,24 @@
+extern crate byteorder;
+
+#[path = "tests.rs"]
+#[cfg(test)]
+mod tests;
+
+#[path = "benchmarks.rs"]
+#[cfg(all(test, feature = "bench_private"))]
+mod benchmarks;
+
+#[path = "v2_serializer.rs"]
+mod v2_serializer;
+pub use self::v2_serializer::{V2Serializer, SerializeError};
```

Comment:
Yeah, I think it will need to be `V2SerializerError` unfortunately because it has error cases that, for instance, V0 does not because that didn't have variable-length encoded ints. That may well be the undoing of being able to genericize over serializers; we'll see.